### ISSUE TITLE
[BUG] materializing a string column from a sliced string view is broken

### ISSUE THREAD

**Issue by abellina** [rapidsai/cudf#3885]
**Describe the bug**
When materializing a `cudf::column` from a sliced string column view, it fails for anything but the first slice (offset 0). This is in `branch-0.12`.

**Steps/Code to reproduce bug**
```
[ RUN      ] SliceStringTest.StringWithNulls2
comparing partition: 0 2 0
comparing COL partition: 0 2 0
comparing partition: 1 3 2
comparing COL partition: 1 1 0
comparing partition: 2 3 1
unknown file: Failure
C++ exception with description "cuDF failure at: /home/abellina/work/cudf/cpp/src/strings/copying/copying.cu:53: Invalid start parameter value." thrown in the test body.
[  FAILED  ] SliceStringTest.StringWithNulls2 (1 ms)

```
The _COL_ row is what happens when we materialize a `cudf::column` from the sliced view.

The issue is with partition 1 and partition 2. 
- Partition 1 had 3 valid items and 2 nulls, but when copied ended up with 1 valid item and 0 nulls.
- Partition 2 fails to materialize.

**Expected behavior**
```
[ RUN      ] SliceStringTest.StringWithNulls2
comparing partition: 0 2 0
comparing COL partition: 0 2 0
comparing partition: 1 3 2
comparing COL partition: 1 3 2
comparing partition: 2 3 1
comparing COL partition: 2 3 1
[       OK ] SliceStringTest.StringWithNulls2 (1 ms)
```

Will PR fix shortly, but this commit should do it: https://github.com/rapidsai/cudf/commit/941926212ab8a7d32ede48cb9ad9198f5ae00cdf

I am not  an expert in what ripple effects this change in `column.cu` might have. @davidwendt @jrhemstad @harrism?

**Comment by davidwendt** [comment]
@abellina You fix here looks correct to me. 
The input `column_view` is already _sliced_ (it already had offset and size) so I believe the original code was causing the offset to happen twice. Your change here fixes that.

**Comment by abellina** [comment]
Yeah the gather is also taking this into account. OK thanks for the feedback @davidwendt

**Comment by kkraus14** [comment]
Fixed by #3886
